### PR TITLE
feat(ui-react-native): Add Verify user subcomponents

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

* Add Verify user subcomponent
* Add Confirm verify user subcomponent
* Add DefaultFormfields
* Add tests

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/ConfirmVerifyUser.tsx [hunk 1]
```diff
@@ -1,17 +1,43 @@
...
+  Footer,
+  Header,
+  isPending,
+}) => {
+  return (
+    <>
+      <Header>{getAccountRecoveryInfoText()}</Header>
+      <FormFields isPending={isPending} fields={fields} />
+      {error ? <ErrorMessage>{error}</ErrorMessage> : null}
+
```
Comment (calebpollman) [inline packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/ConfirmVerifyUser.tsx:25]: Nit: whitespace in JSX

File: packages/react-native/src/primitives/PhoneNumberField/PhoneNumberField.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ export default function PhoneNumberField({
   );
 
   const pickerItems = useMemo(() => {
-    return dialCodes.map((dialCode) => {
+    return dialCodes?.map((dialCode) => {
```
Comment (calebpollman) [inline packages/react-native/src/primitives/PhoneNumberField/PhoneNumberField.tsx:31]: Should we not render the dial code picker in the scenario of no provided `dialCodes`?

**Review by calebpollman — COMMENTED** [review]
Subcomponents are good to go. Left some minor feedback around the `FormFields`

File: packages/react-native/src/Authenticator/common/DefaultFormFields/DefaultFormFields.tsx [hunk 3]
```diff
@@ -1,8 +1,51 @@
...
+          ? PasswordField
+          : type === 'phone'
+          ? PhoneNumberField
+          : TextField;
+
+      return (
+        <Fragment key={name}>
+          <Field
+            {...field}
+            autoCapitalize="none"
```
[the remark this answers is not in the document]
Comment (ioanabrooks) [reply packages/react-native/src/Authenticator/common/DefaultFormFields/DefaultFormFields.tsx]: Done! Making a note to add more tests around it in follow-up.

File: packages/react-native/src/primitives/PhoneNumberField/PhoneNumberField.tsx (hunk 2 above)
Comment (ioanabrooks) [reply packages/react-native/src/primitives/PhoneNumberField/PhoneNumberField.tsx:31]: Addressed, and added test.

File: examples/react-native/storybook/stories/ConfirmVerifyUser.stories.tsx [hunk 4]
```diff
@@ -1,11 +1,35 @@
...
+  label: 'Username',
+  placeholder: 'Username',
+  type: 'default' as const,
+};
+
+const fields = [username];
+
+const props = {
+  fields,
+  Footer: Authenticator.SignIn.Footer,
```
Comment (joebuono) [inline examples/react-native/storybook/stories/ConfirmVerifyUser.stories.tsx:18]: nit: Should be `Authenticator.ConfirmVerifyUser.Footer`

File: packages/react-native/src/Authenticator/Defaults/ConfirmResetPassword/__tests__/ConfirmResetPassword.spec.tsx [hunk 5]
```diff
@@ -2,11 +2,33 @@ import React from 'react';
...
+};
+
+const confirmPassword = {
+  name: 'confirmPassword',
+  label: 'Confirm Password',
+  placeholder: 'Confirm Password',
+  type: 'password' as const,
+};
+
+const fields = [code, newPassword, confirmPassword];
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/Defaults/ConfirmResetPassword/__tests__/ConfirmResetPassword.spec.tsx:28]: Thanks for adding these fields to the tests 👍

File: examples/react-native/storybook/stories/ConfirmVerifyUser.stories.tsx (hunk 4 above)
Comment (ioanabrooks) [reply examples/react-native/storybook/stories/ConfirmVerifyUser.stories.tsx]: copy-pasta ftw. great catch, thanks!

File: packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/ConfirmVerifyUser.tsx [hunk 6]
```diff
@@ -1,17 +1,42 @@
...
+      <Header>{getAccountRecoveryInfoText()}</Header>
+      <FormFields isPending={isPending} fields={fields} />
+      {error ? <ErrorMessage>{error}</ErrorMessage> : null}
+      <Button
+        style={styles.buttonSecondary}
+        textStyle={styles.buttonSecondaryLabel}
+      >
+        {getSkipText()}
+      </Button>
+      <Footer />
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/ConfirmVerifyUser.tsx:31]: Just curious if there's no secondary button for this screen?

File: packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/__tests__/ConfirmVerifyUser.spec.tsx [hunk 7]
```diff
@@ -1,22 +1,43 @@
...
+  it('renders as expected with errors', () => {
+    const error = 'Something went wrong';
+    const { toJSON, getByRole, getByText } = render(
+      <ConfirmVerifyUser {...props} error={error} />
+    );
+    expect(toJSON()).toMatchSnapshot();
+
+    expect(getByRole('alert')).toBeDefined();
+    expect(getByText(error)).toBeDefined();
   });
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/__tests__/ConfirmVerifyUser.spec.tsx:42]: Should we test that the button handle `skipVerification`?

File: packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/ConfirmVerifyUser.tsx [hunk 8]
```diff
@@ -1,17 +1,42 @@
...
+    </>
+  );
 };
 
+const FormFields: DefaultConfirmVerifyUserComponent['FormFields'] = () => null;
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/ConfirmVerifyUser.tsx:36]: Can we just use `DefaultFormFields` here?

File: packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/ConfirmVerifyUser.tsx [hunk 9]
```diff
@@ -1,17 +1,42 @@
...
+
 ConfirmVerifyUser.Footer = DefaultFooter;
-ConfirmVerifyUser.FormFields = DefaultFormFields;
+ConfirmVerifyUser.FormFields = FormFields;
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/ConfirmVerifyUser.tsx:39]: Why can't we just use `DefaultFormFields` here?

File: packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/styles.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
+import { StyleSheet, TextStyle, ViewStyle } from 'react-native';
+
+// TODO: clean up with theming
+interface VerifyUserStyle {
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/styles.ts:4]: nit: Should be `ConfirmVerifyUserStyle`

File: packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/ConfirmVerifyUser.tsx [hunk 11]
```diff
@@ -1,17 +1,42 @@
...
+  Footer,
+  Header,
+  isPending,
+}) => {
+  return (
+    <>
+      <Header>{getAccountRecoveryInfoText()}</Header>
+      <FormFields isPending={isPending} fields={fields} />
+      {error ? <ErrorMessage>{error}</ErrorMessage> : null}
+      <Button
```
[the remark this answers is not in the document]
Comment (ioanabrooks) [reply packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/ConfirmVerifyUser.tsx:26]: yup! thanks. added unit test as well.

File: packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/__tests__/ConfirmVerifyUser.spec.tsx (hunk 7 above)
Comment (ioanabrooks) [reply packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/__tests__/ConfirmVerifyUser.spec.tsx:42]: added!

File: packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/ConfirmVerifyUser.tsx (hunk 6 above)
Comment (ioanabrooks) [reply packages/react-native/src/Authenticator/Defaults/ConfirmVerifyUser/ConfirmVerifyUser.tsx:33]: Didn't look like it to me, double checked with the react web version as well.

File: packages/react-native/src/Authenticator/common/DefaultFormFields/FieldErrors.tsx [hunk 12]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Text } from 'react-native';
+
+import { FieldErrorsProps } from './types';
+
+export const FieldErrors = ({
+  errors,
+  style,
+}: FieldErrorsProps): JSX.Element | null => {
+  if (!(errors?.length > 0)) {
```
Comment (joebuono) [inline packages/react-native/src/Authenticator/common/DefaultFormFields/FieldErrors.tsx:10]: nit: could it just be `!errors?.length`?

**Review by joebuono — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**